### PR TITLE
Make `len()` work with lists

### LINKED ISSUE #925: Implement `ListLen` in the LLVM backend
```py
from ltypes import i32, f64

def test():
    x: list[f64] = [1.0, 2.0]

    i: i32
    for i in range(len(x)):
        print(x[i])

test()
```
```console
lpython examples/expr2.py --new-parser
Internal Compiler Error: Unhandled exception
Traceback (most recent call last):
  Binary file "/home/thirumalai/Open_Source/lpython/src/bin/lpython", in _start()
  File "./csu/../csu/libc-start.c", line 392, in __libc_start_main_impl()
  File "./csu/../sysdeps/nptl/libc_start_call_main.h", line 58, in __libc_start_call_main()
  File "/home/thirumalai/Open_Source/lpython/src/bin/lpython.cpp", line 1081, in ??
    err = compile_python_to_object_file(arg_file, tmp_o, runtime_library_dir, lpython_pass_manager, compiler_options, time_report);
...
...
    this->visit_expr(*x);
  File "/home/thirumalai/Open_Source/lpython/src/libasr/../libasr/asr.h", line 3764, in LFortran::ASR::BaseVisitor<LFortran::ASRToLLVMVisitor>::visit_expr(LFortran::ASR::expr_t const&)
    void visit_expr(const expr_t &b) { visit_expr_t(b, self()); }
  File "/home/thirumalai/Open_Source/lpython/src/libasr/../libasr/asr.h", line 3586, in ??
    case exprType::ListLen: { v.visit_ListLen((const ListLen_t &)x); return; }
LCompilersException: visit_ListLen() not implemented
```

### PR DESCRIPTION
**Pull request by namannimmo10**
Fixes #925

### CODE REVIEW

**Comment by namannimmo10** [comment]
I am also going to do this for tuples.

**Comment by Thirumalai-Shaktivel** [comment]
Perfect, I was also planning to do the same.

File: integration_tests/test_builtin_len.py [hunk 1]
```diff
@@ -7,4 +9,17 @@ def test_len():
...
+    l2: list[f64]
+    l2 = [1.2, 3.4, 5.6, 7.8, 9.0]
+    assert len(l2) == 5
+
+    l3: list[i32] = []
+    l3.append(1)
+    l3.append(2)
+    l3.append(3)
+    assert len(l3) == 3
+
```
Comment (Thirumalai-Shaktivel) [inline integration_tests/test_builtin_len.py:24]: Can you please add this test as well?
```py
    list_len: i32 = len([1.0, 2.0])
    assert list_len == 2
```

File: src/libasr/codegen/asr_to_llvm.cpp [hunk 2]
```diff
@@ -1234,6 +1234,15 @@ class ASRToLLVMVisitor : public ASR::BaseVisitor<ASRToLLVMVisitor>
         tmp = list_api->read_item(plist, pos);
     }
 
+    void visit_ListLen(const ASR::ListLen_t& x) {
+        uint64_t ptr_loads_copy = ptr_loads;
+        ptr_loads = 0;
+        this->visit_expr(*x.m_arg);
+        ptr_loads = ptr_loads_copy;
+        llvm::Value* plist = tmp;
+        tmp = list_api->len(plist);
```
Comment (Thirumalai-Shaktivel) [inline src/libasr/codegen/asr_to_llvm.cpp:1243]: `list_len: i32 = len([1.0, 2.0])` for this example, why not use the value from `x.m_value`??
```suggestion
        if (x.m_value) {
            this->visit_expr(*x.m_value);
        } else {
            uint64_t ptr_loads_copy = ptr_loads;
            ptr_loads = 0;
            this->visit_expr(*x.m_arg);
            ptr_loads = ptr_loads_copy;
            llvm::Value* plist = tmp;
            tmp = list_api->len(plist);
        }
```

File: src/libasr/codegen/llvm_utils.cpp [hunk 3]
```diff
@@ -246,6 +246,10 @@ namespace LFortran {
         return LLVM::CreateLoad(*builder, element_ptr);
     }
 
+    llvm::Value* LLVMList::len(llvm::Value* list) {
+        return LLVM::CreateLoad(*builder, get_pointer_to_current_capacity(list));
```
Comment (czgdp1807) [inline src/libasr/codegen/llvm_utils.cpp:250]:
```suggestion
        return LLVM::CreateLoad(*builder, get_pointer_to_current_end_point(list));
```

**Review by czgdp1807 — CHANGES_REQUESTED** [review]

File: integration_tests/test_builtin_len.py [hunk 4]
```diff
@@ -7,4 +9,20 @@ def test_len():
...
+    l = [1, 2, 3, 4]
+    assert len(l) == 4
+    l2: list[f64]
+    l2 = [1.2, 3.4, 5.6, 7.8, 9.0]
+    assert len(l2) == 5
+
+    l3: list[i32] = []
+    l3.append(1)
+    l3.append(2)
+    l3.append(3)
```
Comment (czgdp1807) [inline integration_tests/test_builtin_len.py:22]: Let's do 20-30 appends in a loop and then check if the `len(xyz_list)` gives correct output.

**Review by czgdp1807 — CHANGES_REQUESTED** [review]

**Review by Thirumalai-Shaktivel — APPROVED** [review]
LGTM! Thank you

**Comment by namannimmo10** [comment]
@czgdp1807 - is this ready?

**Review by czgdp1807 — APPROVED** [review]

**Pull request merged**